src/freenet/store/saltedhash/CipherManager.java

```diff
@@ -61,6 +63,8 @@ protected boolean removeEldestEntry(Map.Entry<ByteArrayWrapper, byte[]> eldest)
 			return size() > 128;
 		}
 	};
+	
+	private ReadWriteLock digestRoutingKeyCacheLock = new ReentrantReadWriteLock();
```

Comment:
This should be `final`. And while you’re at it, make the map above `final`, too, please. 🙂